File: 프로그래머스/2/17677. ［1차］ 뉴스 클러스터링/［1차］ 뉴스 클러스터링.swift
func solution(_ str1:String, _ str2:String) -> Int {
    let A = go(str1.lowercased())
    let B = go(str2.lowercased())
    if A.isEmpty && B.isEmpty { return 65536 }
    let (inter, union) = solve(A, B)
    return Int(Double(inter) / Double(union) * 65536)
}

func go(_ str: String) -> [String] {
    var ret = [String]()
    var str = Array(str)
    for i in 0..<(str.count - 1) {
        if str[i].isLetter && str[i + 1].isLetter {
            ret.append(String(str[i]) + String(str[i + 1]))
        }
    }   
    return ret
}

func solve(_ a: [String], _ b: [String]) -> (Int, Int) {
    var cnt: [String: Int] = [:]
    var cnt2: [String: Int] = [:]    
    for aa in a { cnt[aa, default: 0] += 1 }
    for bb in b { cnt2[bb, default: 0] += 1 }

    let sets = Set(a + b)
    var (inter, union) = (0, 0)
    
    for str in sets {
        inter += min(cnt[str, default: 0], cnt2[str, default: 0])
        union += max(cnt[str, default: 0], cnt2[str, default: 0])
    }
    return (inter, union)
}

File: 프로그래머스/2/17677. ［1차］ 뉴스 클러스터링/［1차］ 뉴스 클러스터링.cpp
#include <string>

using namespace std;

const int mod = 65536;
const int T = 676;
int A[T], B[T];

void parse(const string& s, int cnt[]) {
    for(int i = 0; i < (int)s.size()-1; i++) {
        if(isalpha(s[i]) && isalpha(s[i+1])) {
            int c = (tolower(s[i]) - 'a') * 26 + (tolower(s[i+1]) - 'a');
            cnt[c]++;
        }
    }
}

int solution(string str1, string str2) {
    parse(str1, A);
    parse(str2, B);
    int answer = 0;
    int inter = 0, uni = 0;
    
    for(int i = 0; i < T; i++) {
        inter += min(A[i], B[i]);
        uni += max(A[i], B[i]);
    }
    if(uni == 0) return mod;
    return (int)((double)inter / uni * mod);
}